printer/kubectl_diff.go

```diff
@@ -0,0 +1,45 @@
+package printer
+
+import (
+	"bufio"
+	"fmt"
+	"github.com/kubecolor/kubecolor/config"
+	"io"
+	"regexp"
+)
+
+type DiffPrinter struct {
+	Theme *config.Theme
+}
+
+func (p *DiffPrinter) Print(r io.Reader, w io.Writer) {
+	scanner := bufio.NewScanner(r)
+	for scanner.Scan() {
+		line := scanner.Text()
+		if line == "" {
+			fmt.Fprintln(w, line)
+			continue
+		}
+		parsedLine := p.parseLine(line)
+
+		fmt.Fprintln(w, parsedLine)
+	}
+}
+
+func (p *DiffPrinter) parseLine(line string) string {
+	theme := p.Theme.Diff
+	switch {
+	// All only lines, that starts with a single "+"and anything afterward
+	// e.g. + foo
+	// not allowed: ++ foo
+	case regexp.MustCompile(`^\+[^+]+.*$`).MatchString(line):
+		return theme.Added.Render(line)
+	// All only lines, that starts with a single "-" and anything afterward
+	// e.g. - foo
+	// not allowed: -- foo
+	case regexp.MustCompile(`^-[^-]+.*$`).MatchString(line):
+		return theme.Removed.Render(line)
```

Comment:
Here you're recompiling the regex on each line being parsed, which feels unnecessary.

You could extract the compilation to global variables to skip that, however I think using regex here is overkill and instead it would be better to do some simple string comparisons.

I'm a really big fan of the `strings.HasPrefix` function as it's a fast and non-allocating function. So what if you did this?

```suggestion
	case strings.HasPrefix(line, "+") && !strings.HasPrefix(line, "+++"):
		return theme.Added.Render(line)
	case strings.HasPrefix(line, "-") && !strings.HasPrefix(line, "---"):
		return theme.Removed.Render(line)
```

(I removed the comments as the code became a bit more self-explanatory with those functions)

Alternatively you could add color configs for them too and then match it like so:

```suggestion
	case strings.HasPrefix(line, "+++"):
		return theme.PathLive.Render(line)
	case strings.HasPrefix(line, "+"):
		return theme.Added.Render(line)
	case strings.HasPrefix(line, "---"):
		return theme.PathMerged.Render(line)
	case strings.HasPrefix(line, "-"):
		return theme.Removed.Render(line)
```